From 570bd861c727f38f405890df0c27b41348f3b14d Mon Sep 17 00:00:00 2001
From: Sergey Vinokurov <serg.foo@gmail.com>
Date: Sat, 11 Jan 2025 15:47:30 +0000
Subject: [PATCH] Enlarge volume slider

---
 src/ui/mainwindow.ui | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/src/ui/mainwindow.ui b/src/ui/mainwindow.ui
index 8bc901251..f3abfc58a 100644
--- a/src/ui/mainwindow.ui
+++ b/src/ui/mainwindow.ui
@@ -263,19 +263,24 @@
                   </property>
                  </widget>
                 </item>
-                <item>
-                 <widget class="Amarok::VolumeSlider" name="volume">
+                <item alignment="Qt::AlignmentFlag::AlignVCenter">
+                 <widget class="QSlider" name="volume">
                   <property name="sizePolicy">
-                   <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
+                   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
+                  <property name="minimumSize">
+                   <size>
+                    <width>256</width>
+                   </size>
+                  </property>
                   <property name="maximum">
                    <number>100</number>
                   </property>
                   <property name="orientation">
-                   <enum>Qt::Horizontal</enum>
+                   <enum>Qt::Orientation::Horizontal</enum>
                   </property>
                  </widget>
                 </item>
@@ -900,11 +905,6 @@
  </widget>
  <layoutdefault spacing="6" margin="11"/>
  <customwidgets>
-  <customwidget>
-   <class>Amarok::VolumeSlider</class>
-   <extends>QSlider</extends>
-   <header>widgets/sliderwidget.h</header>
-  </customwidget>
   <customwidget>
    <class>AnalyzerContainer</class>
    <extends>QWidget</extends>
-- 
2.47.0

