From 822686254cb8385a11c10d94371beb9309114de4 Mon Sep 17 00:00:00 2001
From: Sergey Vinokurov <serg.foo@gmail.com>
Date: Sat, 11 Jan 2025 15:41:35 +0000
Subject: [PATCH 1/2] Enlarge playback control buttons

---
 src/ui/mainwindow.ui | 38 ++++++++++++++++++++++++++++----------
 1 file changed, 28 insertions(+), 10 deletions(-)

diff --git a/src/ui/mainwindow.ui b/src/ui/mainwindow.ui
index a2663badc..8bc901251 100644
--- a/src/ui/mainwindow.ui
+++ b/src/ui/mainwindow.ui
@@ -134,10 +134,16 @@
                 </property>
                 <item>
                  <widget class="QToolButton" name="back_button">
+                  <property name="minimumSize">
+                   <size>
+                    <width>48</width>
+                    <height>48</height>
+                   </size>
+                  </property>
                   <property name="iconSize">
                    <size>
-                    <width>22</width>
-                    <height>22</height>
+                    <width>48</width>
+                    <height>48</height>
                    </size>
                   </property>
                   <property name="autoRaise">
@@ -147,10 +153,16 @@
                 </item>
                 <item>
                  <widget class="QToolButton" name="pause_play_button">
+                  <property name="minimumSize">
+                   <size>
+                    <width>48</width>
+                    <height>48</height>
+                   </size>
+                  </property>
                   <property name="iconSize">
                    <size>
-                    <width>22</width>
-                    <height>22</height>
+                    <width>48</width>
+                    <height>48</height>
                    </size>
                   </property>
                   <property name="autoRaise">
@@ -163,10 +175,16 @@
                   <property name="enabled">
                    <bool>false</bool>
                   </property>
+                  <property name="minimumSize">
+                   <size>
+                    <width>48</width>
+                    <height>48</height>
+                   </size>
+                  </property>
                   <property name="iconSize">
                    <size>
-                    <width>22</width>
-                    <height>22</height>
+                    <width>48</width>
+                    <height>48</height>
                    </size>
                   </property>
                   <property name="popupMode">
@@ -181,14 +199,14 @@
                  <widget class="QToolButton" name="forward_button">
                   <property name="minimumSize">
                    <size>
-                    <width>42</width>
-                    <height>29</height>
+                    <width>48</width>
+                    <height>48</height>
                    </size>
                   </property>
                   <property name="iconSize">
                    <size>
-                    <width>22</width>
-                    <height>22</height>
+                    <width>48</width>
+                    <height>48</height>
                    </size>
                   </property>
                   <property name="popupMode">
-- 
2.47.0

