--- a/src/VBox/Devices/PC/ipxe/Makefile.kmk	2023-09-22 12:31:10.465112117 +0100
+++ b/src/VBox/Devices/PC/ipxe/Makefile.kmk	2023-09-22 12:31:53.328755669 +0100
@@ -50,3 +50,3 @@
 	$(call MSG_TOOL,bin2c,iPxeBiosBin,$<,$@)
-	$(QUIET)$(VBOX_BIN2C) -min 32 -max 56 -mask 0x1ff -export NetBiosBinary $< $@
+	$(QUIET)$(VBOX_BIN2C) -min 32 -max 58 -mask 0x1ff -export NetBiosBinary $< $@
 
@@ -60,3 +60,3 @@
 	$(call MSG_TOOL,bin2c,iPxeBiosBin,$<,$@)
-	$(QUIET)$(VBOX_BIN2C) -min 32 -max 56 -export NetBiosBinary $< $@
+	$(QUIET)$(VBOX_BIN2C) -min 32 -max 58 -export NetBiosBinary $< $@
 
